web/src/engine/osk/src/input/gestures/specsForKeyboard.ts

```diff
@@ -0,0 +1,349 @@
+import {
+  gestures,
+  GestureModelDefs
+} from '@keymanapp/gesture-recognizer';
+
+import {
+  Codes,
+  Keyboard
+} from '@keymanapp/keyboard-processor';
+
+import { type KeyElement } from '../../keyElement.js';
+
+import specs = gestures.specs;
+
+/**
+ * Defines the set of gestures appropriate for use with the specified Keyman keyboard.
+ * @param keyboard
+ * @returns
+ */
+export function modelSetForKeyboard(keyboard: Keyboard): GestureModelDefs<KeyElement> {
+  // TODO:  keyboard-specific config stuff
+  // if `null`, assume a no-flick keyboard (assuming our default layout has no flicks)
+
+  // Idea:  if we want to get fancy, we could detect if the keyboard even _supports_ some of
+  // the less common gestures and just... not include the model if it doesn't.  Should only
+  // do that if it's computationally "cheap", though.
+  return {
+    gestures: [
+      // TODO:  some, if not all, will probably utilize methods, rather than constant definitions.
+      // But, this should be fine for a first-pass integration attempt.
+      LongpressModel,
+      MultitapModel,
+      SimpleTapModel,
+      SubkeySelectModel,
+      ModipressStartModel,
+      ModipressEndModel
+    ],
+    sets: {
+      default: [LongpressModel.id, SimpleTapModel.id, ModipressStartModel.id],
+      modipress: [LongpressModel.id, SimpleTapModel.id], // no nested modipressing
+      none: []
+    }
+  }
+}
+
+// #region Definition of models for paths comprising gesture-stage models
+
+type ContactModel = specs.ContactModel<KeyElement>;
+
+export const InstantContactRejectionModel: ContactModel = {
+  itemPriority: 0,
+  pathResolutionAction: 'reject',
+  pathModel: {
+    evaluate: (path) => 'resolve'
+  }
+}
+
+export const InstantContactResolutionModel: ContactModel = {
+  itemPriority: 0,
+  pathResolutionAction: 'resolve',
+  pathModel: {
+    evaluate: (path) => 'resolve'
+  }
+}
+
+export const LongpressDistanceThreshold = 10;
+export const MainContactLongpressSourceModel: ContactModel = {
+  itemChangeAction: 'reject',
+  itemPriority: 0,
+  pathResolutionAction: 'resolve',
+  timer: {
+    duration: 500,
```

Comment:
Is this the only place this magic is used?

Reply:
This gets handled far better within #9698.  It... might be the only place within this PR, but I'm unsure of that.

From said PR:

https://github.com/keymanapp/keyman/blob/f67549a3e445278e0918484f40b078d928e0f0f6/web/src/engine/osk/src/input/gestures/specsForLayout.ts#L51-L58